code/src/components/Input.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { tasks } from "reducers/tasks";
+
+import styled from "styled-components";
+
+let taskId = 0;
+
+export const Input = ({ handleClick, handleSubmit }) => {
+  let [todoText, setTodoText] = useState("");
+
+  const dispatch = useDispatch();
+
+  handleClick = () => {
+    setTodoText("");
+    dispatch(
+      tasks.actions.addTask({
+        text: todoText,
+        id: taskId++
+      })
+    );
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+  };

```

Comment:
If you want to make it even more user friendly, so that you can just press enter when typed the todo instead of having to use the button, you could do line 15-21 inside the handleSubmit function instead. And then have the button onClick and the onSubmit calling that function.
```
 handleSubmit = e => {
    e.preventDefault();
    setTodoText("");
    dispatch(
      tasks.actions.addTask({
        text: todoText,
        id: taskId++
      })
    );
  };
```

Reply:
💪